programs/bpf_loader/src/syscalls/cpi.rs

```diff
@@ -1128,29 +1359,39 @@ fn update_caller_account(
     *caller_account.lamports = callee_account.get_lamports();
     *caller_account.owner = *callee_account.get_owner();
 
-    if direct_mapping && caller_account.original_data_len > 0 {
-        // Since each instruction account is directly mapped in a memory region
-        // with a *fixed* length, upon returning from CPI we must ensure that the
-        // current capacity is at least the original length (what is mapped in
-        // memory), so that the account's memory region never points to an
-        // invalid address.
-        let min_capacity = caller_account.original_data_len;
-        if callee_account.capacity() < min_capacity {
-            callee_account.reserve(min_capacity.saturating_sub(callee_account.capacity()))?
-        }
+    let mut zero_all_spare_capacity = false;
```

Comment:
nit: actually not _all_; rather `should_zero_mapped_spare_capacity`.

Reply:
or just `should_zero_spare_capacity` (drop the overly qualifiying _all_); seems `spare` word already meaning `mapped_capacity`